src/ZLogger.Unity/Assets/Scripts/ZLogger/AsyncStreamLineMessageWriter.cs

```diff
@@ -1,62 +1,212 @@
+                            log.LogInfo.Timestamp,
+                            log.LogInfo.LogLevel,
+                            log.LogInfo.EventId,
+                            log.LogInfo.Exception
+                        );
+    
+                        bufferWriterForDebugSpamMsg.Clear();
+                        bufferWriterForDebugSpamMsg.Write(Encoding.UTF8.GetBytes("\""));
+                        log.FormatUtf8(bufferWriterForDebugSpamMsg, new(), null);
+                        bufferWriterForDebugSpamMsg.Write(Encoding.UTF8.GetBytes("\""));
+                        string logStr = Encoding.UTF8.GetString(bufferWriterForDebugSpamMsg.WrittenSpan);
+    
+                        IZLoggerEntry? entry =
+                            FormatLogState<object, int, int, int, int, int, int, int, bool, int, string>.Factory(
+                                new(
+                                    null,
+                                    "LOG DROPPED {9}, postTimes.Count {8} did Drop: {7}. Dropping: ({0}) Trace. ({1}) Debug. ({2})  Information. ({3})  Warning. ({4})  Error. ({5})  Critical. ({6})  None",
+                                    dropSummary[LogLevel.Trace],
+                                    dropSummary[LogLevel.Debug],
+                                    dropSummary[LogLevel.Information],
+                                    dropSummary[LogLevel.Warning],
+                                    dropSummary[LogLevel.Error],
+                                    dropSummary[LogLevel.Critical],
+                                    dropSummary[LogLevel.None],
+                                    didDrop,
+                                    postTimesQ.Count,
+                                    logStr
+                                ),
+                                logInfo
+                            );
+    
+                        lock (lockObject) channel.Writer.TryWrite(entry);
+                    }
+    
+                    dropSummary[log.LogInfo.LogLevel]++;
+                    lock (lockObject) didDrop = true;
+                }
+                else
+                {
+                    bool tryWrite;
+                    lock (lockObject) tryWrite = channel.Writer.TryWrite(log);
+
+                    if (tryWrite)
+                        postTimesQ.Enqueue(log.LogInfo.Timestamp);
+                }
+            }
+        }
+
+        private bool CheckPostsTimesAndSetIsSpamming()
+        {
+            // fixes the issue where seeing summary without the actual logs
+            // maybe we can put the lock further down the PostTimesCheck method
+            lock (lockObject)
+            {
+                bool? spamming = PostsTimesCheck();
+                if (spamming.HasValue)
+                    lock (lockObject)
+                        isSpamming = spamming.Value;
+
+                return isSpamming;
+            }
+        }
+
+        private bool? PostsTimesCheck()
```

Comment:
maybe document it a bit (bool -> null means don't change current status, etc.)

Reply:
a700c1c - minor document function